pysatMadrigal/instruments/methods/madrigal.py

```diff
@@ -58,114 +55,226 @@ def load(fnames, tag=None, inst_id=None, xarray_coords=[]):
-        + data.loc[:, 'sec']
-    time = pysat.utils.time.create_datetime_index(year=data.loc[:, 'year'],
-                                                  month=data.loc[:, 'month'],
-                                                  day=data.loc[:, 'day'],
-                                                  uts=uts)
-    # Declare index or recast as xarray
-    if len(xarray_coords) > 0:
-        if not np.all([xkey.lower() in data.columns
-                       for xkey in xarray_coords]):
-            estr = 'unknown coordinate key in {:}, '.format(xarray_coords)
-            estr += 'use only {:}'.format(data.columns)
-            raise ValueError(estr)
-
-        # Append time to the data frame and add as the first coordinate
-        data = data.assign(time=pds.Series(time, index=data.index))
-        xarray_coords.insert(0, 'time')
-
-        # Set the indices
-        data = data.set_index(xarray_coords)
-
-        # Recast the data as an xarray
-        data = data.to_xarray()
-    else:
-        # Set the index to time, and put up a warning if there are duplicate
-        # times.  This could mean the data should be stored as an xarray
-        # DataSet
-        data.index = time
-
-        if np.any(time.duplicated()):
-            logger.warning(' '.join(["duplicated time indices, consider",
-                                     "specifing additional coordinates and",
-                                     "storing the data as an xarray DataSet"]))
+    fdata = []
+
+    # Load the file data
+    for fname in fnames:
+        if file_type == "netCDF4":
+            # Xarray natively opens netCDF data into a Dataset
+            file_data = xr.open_dataset(fname)
+
+            # Currently not saving file header data, as all metadata is at
+            # the data variable level
+            if len(labels) == 0:
+                for item in file_data.data_vars.keys():
+                    name_string = item
+                    unit_string = file_data[item].attrs['units']
+                    desc_string = file_data[item].attrs['description']
+                    labels.append(name_string)
+                    meta[name_string.lower()] = {'long_name': name_string,
+                                                 'units': unit_string,
+                                                 'desc': desc_string}
+                    # remove any metadata from xarray
+                    file_data[item].attrs = {}
+
+            # Reset UNIX timestamp as datetime and set it as an index
+            file_data = file_data.rename({'timestamps': 'time'})
+            time_data = pds.to_datetime(file_data['time'], unit='s')
+            data = file_data.assign_coords({'time': ('time', time_data)})
+
+        elif file_type == "hdf5":
+            # Open the specified file and get the data and metadata
+            filed = h5py.File(fname, 'r')
+            file_data = filed['Data']['Table Layout']
```

Comment:
This line is failing for me using the TEC instrument. `filed` has variable names for keys. Neither `Data` nor `Metadata` at the `filed` level. While I was able to find data in this new `filed`, I wasn't able to find where the metadata is.

I'm trying out JRO.

Reply:
This may be another version issue >_<

Reply:
Could be. I was able to load JRO data without a problem after your fixes over in pysat. I was not able to load the gnss data in hdf format. What version of h5py are you using?
```
In [7]: h5py.__version__
Out[7]: '2.10.0'
```

Reply:
I have the same version...

Reply:
I think I sorted out where the issue pops up. If I only work with hdf files, or only netcdf files, things work. However, if I say have hdf files, then download netcdf files, then the original hdf files no longer work. I think the same the other way as well. As long as folks stick with only one file type then it works, but there is potential for users to break things down the road.

I'm sure you could fix it, but since the hdfs take much longer to load, you may want to consider going netcdf only for this instrument.

Reply:
> if I say have hdf files, then download netcdf files, then the original hdf files no longer work.

That's so strange!  What would change???  I'm going to try and sort this with JRO data.

Reply:
Ok, what are your thoughts on putting this in as an issue and solving it later?

Reply:
I'm ok with that. Perhaps with a note in a docstring for the interim?

Didn't you say on a recent call that the madrigal files always come down as .hdf and you rename them?

Reply:
That sounds good to me.

And yes.  You can look in the madrigal download routine to see what's up there.

Reply:
Created issue #25 and added a note to the docstring.